src/schema/users.ts

```diff
@@ -1245,6 +1261,50 @@ export const resolvers: IResolvers<unknown, BaseContext> = traceResolvers<
 
       return streak;
     },
+    streakRecover: async (
+      _,
+      __,
+      ctx: AuthContext,
+    ): Promise<StreakRecoverQueryResult> => {
+      const { userId } = ctx;
+
+      const streak = await ctx.con.getRepository(UserStreak).findOneBy({
+        userId,
+      });
+      const timeForRecoveryPassed = streak.currentStreak > 1;
+
+      const key = generateStorageKey(
+        StorageTopic.Streak,
+        StorageKey.Reset,
+        userId,
+      );
+      const oldStreakLength = Number(await getRedisObject(key));
```

Comment:
We can get `streak` and `oldStreakLength` with `Promise.all` in parallel?

Reply:
Updated! ✅